suite-native/discovery/src/discoverySelectors.ts

```diff
@@ -45,3 +71,84 @@ export const selectDiscoveryAccountsAnalytics = (
             };
         }),
     );
+
+export const selectNetworksWithUnfinishedDiscovery = (
+    state: DeviceRootState &
+        AccountsRootState &
+        FeatureFlagsRootState &
+        DiscoveryConfigSliceRootState,
+    areTestnetsEnabled: boolean,
+) => {
+    const enabledNetworkSymbols = selectEnabledDiscoveryNetworkSymbols(state);
+    const accounts = selectDeviceAccounts(state);
+    const supportedNetworks = selectDiscoverySupportedNetworks(state, areTestnetsEnabled);
+    const isCoinEnablingActive = selectIsFeatureFlagEnabled(
+        state,
+        FeatureFlag.IsCoinEnablingActive,
+    );
+
+    const enabledNetworks = supportedNetworks.filter(n =>
+        isCoinEnablingActive ? enabledNetworkSymbols.includes(n.symbol) : true,
+    );
+
+    return getNetworksWithUnfinishedDiscovery(enabledNetworks, accounts, ACCOUNTS_LIMIT);
+};
+
+//we should run discovery when there are network symbols with unfinished discovery
+export const selectShouldRunDiscoveryForDevice = (
+    state: DeviceRootState &
+        AccountsRootState &
+        FeatureFlagsRootState &
+        DiscoveryConfigSliceRootState,
+) => {
+    const isPortfolioTrackerDevice = selectIsPortfolioTrackerDevice(state);
+    if (isPortfolioTrackerDevice) {
+        return false;
+    }
+
+    const areTestnetsEnabled = selectAreTestnetsEnabled(state);
+    const networksWithUnfinishedDiscovery = selectNetworksWithUnfinishedDiscovery(
+        state,
+        areTestnetsEnabled,
+    );
+
+    return networksWithUnfinishedDiscovery.map(n => n.symbol).length > 0;
+};
+
+// we do not run discovery for unsupported device (e.g. old firmware, portfolio device, unauthorized), when discovery is already running or when device is in view-only mode
+export const selectCanRunDiscoveryForDevice = (
+    state: DeviceRootState &
+        AccountsRootState &
+        DiscoveryRootState &
+        FeatureFlagsRootState &
+        DiscoveryConfigSliceRootState,
+) => {
+    const deviceState = selectDeviceState(state);
+    if (!deviceState) {
+        return false;
+    }
+
+    const discovery = selectDeviceDiscovery(state);
+    const deviceModel = selectDeviceModel(state);
+    const deviceFwVersion = selectDeviceFirmwareVersion(state);
+    const isDeviceAuthorized = selectIsDeviceAuthorized(state);
+    const isPortfolioTrackerDevice = selectIsPortfolioTrackerDevice(state);
+
+    const isDeviceFirmwareVersionSupported = isFirmwareVersionSupported(
+        deviceFwVersion,
+        deviceModel,
+    );
+
+    const isDeviceInViewOnlyMode = selectIsDeviceInViewOnlyMode(state);
+    const isDeviceUnlocked = selectIsDeviceUnlocked(state);
+
+    const canRunDiscovery =
+        !discovery &&
+        isDeviceAuthorized &&
+        !isPortfolioTrackerDevice &&
+        !isDeviceInViewOnlyMode &&
+        isDeviceUnlocked &&
```

Comment:
Not sure if it it's correct condition. If device is locked, we should provide a way to unlock and run discovery, shouldn't we?

Reply:
Correct, but imo this should happen elsewhere. Not in the discovery check